app/main.py

```diff
@@ -30,19 +30,20 @@ def create_knights(knights: dict) -> {Knight}:
     return knights_dict
 
 
+def duel(knight1: Knight, knight2: Knight) -> None:
+    knight1.hit(knight2)
+    knight2.hit(knight1)
+
+
 def battle(knights_config: dict) -> {}:
     knights = create_knights(knights_config)
     [knight.prepare_for_battle() for knight in knights.values()]
-    lancelot = knights["Lancelot"]
-    mordred = knights["Mordred"]
-    arthur = knights["Artur"]
-    red_knight = knights["Red Knight"]
-
-    lancelot.hit(mordred)
-    mordred.hit(lancelot)
-
-    arthur.hit(red_knight)
-    red_knight.hit(arthur)
+    participants = {
+        knights["Lancelot"]: knights["Mordred"],
+        knights["Artur"]: knights["Red Knight"]
+    }
```

Comment:
these methods look great, but the names of participants are still hard-coded inside battle method. How about making it receive these names say in form of list of tuples(or dicts or else) so in case when you need to change dueling pairs you just pass different tuple and dont have to rewrite anything in battle method?

Reply:
I get a dict from ```create_knights``` method in next structure: ```{name_of_knight: knight_object}```
after I get objects by name in ```participants```
as I understand do I need to change ```participants``` variable which need to have just 2 tuple of names(str)?

Reply:
i would suggest just to add another parameter to battle() func after knights_config, something that you could iterate through like "(knight_one, knight_two)" so that another dueling pair would be taken just by passing these variables ( " knights[knight_one]: knights[knight_two] ") and so you wouldnt need to have any name like "Lancelot" and other inside battle() at all